### PR TITLE
[Networking] Bump libp2p v0.23.3 -> v0.24.2

### PR DESCRIPTION
**Pull request by SaveTheRbtz**
Updated dependencies:
`https://github.com/libp2p/go-libp2p-tls` -> `github.com/libp2p/go-libp2p/p2p/security/tls`

This is our last dependency on the deprecated `github.com/libp2p/go-libp2p-core`:
```
$ go mod why github.com/libp2p/go-libp2p-core
# github.com/libp2p/go-libp2p-core
(main module does not need package github.com/libp2p/go-libp2p-core)
```

Full release notes: https://github.com/libp2p/go-libp2p/releases/tag/v0.24.0

While here, also removed the `replace mellium.im/sasl => github.com/mellium/sasl v0.2.1`. Module is not used:
```
$ go mod why mellium.im/sasl
# mellium.im/sasl
(main module does not need package mellium.im/sasl)
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -46,14 +46,14 @@ require (
 	github.com/ipfs/go-log v1.0.5
 	github.com/ipfs/go-log/v2 v2.5.1
 	github.com/libp2p/go-addr-util v0.1.0
-	github.com/libp2p/go-libp2p v0.23.3
+	github.com/libp2p/go-libp2p v0.24.0
```
Comment (zhangchiqing) [inline go.mod:49]: Is there any breaking change to us?

**Review by yhassanzadeh13 — DISMISSED** [review]

File: go.mod (hunk 1 above)
Comment (SaveTheRbtz) [reply go.mod:49]: A couple. Mostly tests and transient deps.

**Review by zhangchiqing — APPROVED** [review]

**Review by yhassanzadeh13 — CHANGES_REQUESTED** [review]
We have found some unexpected behavior with this upgrade and shared it with the libp2p team. As discussed with @peterargue and @SaveTheRbtz we block this PR till we have a response from libp2p. More details are available [here](https://www.notion.so/dapperlabs/Upgrade-to-libp2p-v0-24-x-24f7ba7341fe4152b3ca99afb45830ac).

**Review by yhassanzadeh13 — APPROVED** [review]

**Comment by SaveTheRbtz** [comment]
bors merge

**Comment by SaveTheRbtz** [comment]
bors merge

**Comment by SaveTheRbtz** [comment]
bors merge

**Comment by SaveTheRbtz** [comment]
bors merge

**Comment by SaveTheRbtz** [comment]
bors merge

**Pull request merged**